packages/graph/hash_graph/lib/graph/src/knowledge/entity/mod.rs

```diff
@@ -25,12 +29,69 @@ impl fmt::Display for EntityId {
     }
 }
 
+/// An entity.
+///
+/// When expressed as JSON, this should validate against its respective entity type(s).
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Component)]
 pub struct Entity {
     #[serde(flatten)]
     properties: HashMap<BaseUri, serde_json::Value>,
 }
 
+/// A record of an entity that has been persisted in the datastore, with its associated metadata.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Component)]
+pub struct PersistedEntity {
+    inner: Entity,
+    id: EntityId,
+    version: DateTime<Utc>,
+    type_versioned_uri: VersionedUri,
```

Comment:
Might need `#[component(value_type = String)]` on this property
```suggestion
    #[component(value_type = String)]
    type_versioned_uri: VersionedUri,
```

Reply:
That's been added